static/elements/chromedash-gantt.js

```diff
@@ -0,0 +1,206 @@
+      }
+      .diamond span {
+        font-size: 14px;
+        font-weight: bold;
+        display: table-cell;
+        transform: rotate(-45deg);
+        width: 34px;
+        height: 34px;
+        vertical-align: middle;
+      }
+      .offset_0 { padding-left: 0px; }
+      .offset_1 { padding-left: 8px; }
+      .offset_2 { padding-left: 16px; }
+      .offset_3 { padding-left: 24px; }
+    `];
+  }
+
+  firstPhrase(milestone) {
+    if (milestone <= this.stableMilestone) {
+      return 'First milestone with this feature';
+    } else {
+      return 'First expected milestone with this feature';
+    }
+  }
+
+  renderDevTrial(milestone) {
+    if (!milestone) return nothing;
+    return html`
+      <div class="diamond dev_trial
+                  ${milestone == this.stableMilestone ? 'stable' : ''}"
+       title="${this.firstPhrase(milestone)} available to developers behind a flag"
+      ><span>${milestone}</span></div>`;
+  }
+
+  renderOriginTrial(milestone) {
+    if (!milestone) return nothing;
+    return html`
+      <div class="diamond origin_trial
+                  ${milestone == this.stableMilestone ? 'stable' : ''}"
+       title="${this.firstPhrase(milestone)} enabled on participating origins"
+      ><span>${milestone}</span></div>`;
+  }
+
+  renderShipping(milestone) {
+    if (!milestone) return nothing;
+    return html`
+      <div class="diamond shipping
+                  ${milestone == this.stableMilestone ? 'stable' : ''}"
+       title="${this.firstPhrase(milestone)} enabled by default"
+      ><span>${milestone}</span></div>`;
+  }
+
+  renderRow(
+    platform, devTrialMilestone, originTrialMilestone, shippingMilestone,
+    sortedMilestones) {
+    if (!devTrialMilestone && !originTrialMilestone && !shippingMilestone) {
+      return nothing;
+    }
+
+    let dtOffset = sortedMilestones.dt.indexOf(devTrialMilestone);
+    let otOffset = sortedMilestones.ot.indexOf(originTrialMilestone);
+    let shipOffset = sortedMilestones.ship.indexOf(shippingMilestone);
+
+    return html`
+       <ul class="bar ${originTrialMilestone ? 'with-ot' : 'without-ot'}">
+         <li class="platform">${platform}</li>
+         <li class="${'offset_' + dtOffset}">
+           ${this.renderDevTrial(devTrialMilestone)}
+         </li>
+         <li class="${'offset_' + otOffset}">
+           ${this.renderOriginTrial(originTrialMilestone)}
+         </li>
+         <li class="${'offset_' + shipOffset}">
+           ${this.renderShipping(shippingMilestone)}
+         </li>
+       </ul>
+    `;
+  }
+
+  render() {
+    let f = this.feature;
+    let dtDesktop = f.dt_milestone_desktop_start;
+    let otDesktop = f.ot_milestone_desktop_start;
+    let shipDesktop = f.browsers.chrome.desktop;
+    let dtAndroid = f.dt_milestone_android_start;
+    let otAndroid = f.ot_milestone_android_start;
+    let shipAndroid = f.browsers.chrome.android;
+    let dtIos = f.dt_milestone_ios_start;
+    let otIos = null;
```

Comment:
`otIos` and `otWebview` are null but never assigned to a real value in the code

Reply:
I added a comment to explain that those two platforms do not support origin trials.  I defined them as variables rather than using null in the templates below so that the fact that they were always null would be more obvious and the set of variables more consistent.  Let me know if you think using in-line null values would be better.